bittensor/dendrite.py

```diff
@@ -237,22 +263,32 @@ def _handle_request_errors(self, synapse, request_name, exception):
             request_name: The name of the request during which the exception occurred.
             exception: The exception object caught during the request.
 
+        Returns:
+            bittensor.Synapse: The updated synapse object with the error status code and message.
+
         Note:
             This method updates the synapse object in-place.
         """
+
+        self.log_exception(exception)
+
+        error_info = DENDRITE_ERROR_MAPPING.get(type(exception), DENDRITE_DEFAULT_ERROR)
+        status_code, status_message = error_info
+
+        if status_code:
+            synapse.dendrite.status_code = status_code  # type: ignore
+        elif isinstance(exception, aiohttp.ClientResponseError):
+            synapse.dendrite.status_code = str(exception.code)  # type: ignore
```

Comment:
`code` is deprecated for exactions. we need to use `status` instead of this in the future.